src/Microsoft.AspNetCore.Mvc.ViewFeatures/ViewComponents/DefaultViewComponentHelper.cs

```diff
@@ -134,15 +136,25 @@ private ViewComponentDescriptor SelectComponent(Type componentType)
             ViewComponentDescriptor descriptor,
             object arguments)
         {
+            IDictionary<string, object> argumentDictionary;
+
+            var methodParams = descriptor.MethodInfo.GetParameters();
```

Comment:
Seeing `GetParameters()` in this code is a red flag. I don't think we could avoid the `IsAssignableFrom` though :cry: 

should this be one the descriptor?